src/cmlibs/widgets/handlers/fixedaxistranslation.py

```diff
@@ -0,0 +1,156 @@
+        for glyph in (self._glyphs + self._reverse_glyphs):
+            glyph.setVisibilityFlag(True)
+            glyph.setMaterial(self._default_material)
+            set_glyph_position(glyph, centroid)
+
+    def leave(self):
+        for glyph in (self._glyphs + self._reverse_glyphs):
+            glyph.setVisibilityFlag(False)
+
+        self._start_position = None
+        self._selected_index = None
+
+    def mouse_press_event(self, event):
+        super().mouse_press_event(event)
+
+        pixel_scale = self._scene_viewer.get_pixel_scale()
+        x, y = event.x() * pixel_scale, event.y() * pixel_scale
+        self._start_position = [x, y]
+
+        graphic = self._scene_viewer.get_nearest_graphics_point(x, y)
+        if graphic and graphic.isValid():
+            for i in range(len(self._glyphs)):
+                if graphic == self._glyphs[i] or graphic == self._reverse_glyphs[i]:
+                    self._selected_index = i
+                    self._glyphs[i].setMaterial(self._selected_material)
+                    self._reverse_glyphs[i].setMaterial(self._selected_material)
+
+    def mouse_move_event(self, event):
+        if self._selected_index is not None:
+            pixel_scale = self._scene_viewer.get_pixel_scale()
+            x = event.x() * pixel_scale
+            y = event.y() * pixel_scale
+            pos = get_glyph_position(self._glyphs[2])
+            screen_pos = self._scene_viewer.project(pos[0], pos[1], pos[2])
+            global_cur_pos = self._scene_viewer.unproject(x, -y, screen_pos[2])
+            global_old_pos = self._scene_viewer.unproject(self._start_position[0], -self._start_position[1], screen_pos[2])
+            global_pos_diff = sub(global_cur_pos, global_old_pos)
+
+            field_cache = self._glyph_fields[self._selected_index].getFieldmodule().createFieldcache()
+            _, vector = self._glyph_fields[self._selected_index].evaluateReal(field_cache, 3)
+            proj_n = mult(vector, dot(global_pos_diff, vector))
+
+            scene = self._glyphs[2].getScene()
+            scene.beginChange()
+
+            translate_nodes(self._model.get_projection_plane_region(), proj_n)
+            centroid = calculate_centroid(self._model.plane_nodes_coordinates())
+            if centroid is not None:
+                self._model.set_rotation_point(centroid)
+                for glyph in (self._glyphs + self._reverse_glyphs):
+                    set_glyph_position(glyph, centroid)
+
+            scene.endChange()
+            self._start_position = [x, y]
+        else:
+            super(FixedAxisTranslation, self).mouse_move_event(event)
+
+    def mouse_release_event(self, event):
+        super().mouse_release_event(event)
+
+        for glyph in (self._glyphs + self._reverse_glyphs):
+            glyph.setMaterial(self._default_material)
+
+        self._start_position = None
+        self._selected_index = None
+
+
+def calculate_orthogonal_vectors(points):
```

Comment:
If this isn't used externally, start it with an underscore.

Reply:
This has been corrected.